code/modules/chemistry/Reagents-PoisonEtc.dm

```diff
@@ -262,34 +262,56 @@ datum
 			penetrates_skin = 1
 			blob_damage = 5
 			value = 7 // 3 2 1 heat
+			target_organs = list("left_lung","right_lung","heart")
+			flushing_multiplier = 0.8
 			var/counter = 1
 
-			on_mob_life(var/mob/M, var/mult = 1) // -cogwerks. previous version
+			reaction_mob(var/mob/M, var/method=TOUCH, var/volume)
+				. = ..()
+				if (method!=TOUCH)             //is not as effective when entering the skin, takes more time to metabolize
+					counter = max(counter, 12)
+				return
+
+			on_mob_life(var/mob/M, var/mult = 1)
 				if (!M) M = holder.my_atom
 				if (!counter) counter = 1
-				M.take_toxin_damage(1.5 * mult)
-				if (probmult(8))
-					M.emote("drool")
-				if (prob(15))
-					boutput(M, "<span class='alert'>You cannot breathe!</span>")
-					M.losebreath += (1 * mult)
-					M.emote("gasp")
-				switch(counter += (1 * mult))
-					if (20 to 30)
-						if (prob(15))
-							boutput(M, "<span class='alert'>You feel weak.</span>")
-							M.setStatusMin("stunned", 0.5 SECONDS * mult)
-							M.take_toxin_damage(0.5 * mult)
-					if (30 to 45)
-						if (prob(20))
-							boutput(M, "<span class='alert'>You feel very weak.</span>")
-							M.setStatusMin("stunned", 1 SECONDS * mult)
-							M.take_toxin_damage(1 * mult)
-					if (45 to INFINITY)
-						if (prob(25))
-							boutput(M, "<span class='alert'>You feel horribly weak.</span>")
-							M.setStatusMin("stunned", 2 SECONDS * mult)
-							M.take_toxin_damage(1.5 * mult)
+				M.take_toxin_damage(0.5 * mult)  // a bit of tox to show you that something is indeed, not alright
+
+				if (holder.get_reagent_amount("space_drugs") > 5) //amyl nitrate inhibits cyanide's effects IRL, space drugs is pretty close
+					counter += 0.25 * mult
+				else
+					counter += max(0.5, (holder.get_reagent_amount(src.id) ** 0.36 / 1.75) * mult)//speed of poisoning scales nonlinearly with dosage
```

Comment:
shouldn't this mult be outside of your max? a really slow tick will be capped at +0.5 on the counter